northstar/src/runtime/container_key.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 ESRLabs
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+use super::{Name, RepositoryId, Version};
+use serde::{Deserialize, Serialize};
+use std::{
+    convert::TryFrom,
+    fmt::{self, Display},
+    sync::Arc,
+};
+
+#[derive(Clone, Eq, PartialEq, Debug, Hash, Serialize, Deserialize)]
+pub struct ContainerKey {
+    #[serde(flatten)]
+    inner: Arc<Inner>,
```

Comment:
not sure why you make this so complicated...no need to put this in an `Arc`.
A key/an id is also a value object and should be cloned not used as  reference

Reply:
Right it should be cloned - that's why it's done this way to make the clone cheap. Instead of cloning two strings and a version just clone the `Arc`.

Reply:
ok...still a typical example of POITROAE

Reply:
I think it's worth to save the clones since it's anyway not visible to the user of `ContainerKey`.

A clone would clone 2 Strings, 3 u64 (yes that's cheap) and 2 `Vec`s that potentially contain `Strings`.